packages/@aws-cdk/aws-glue-alpha/lib/job.ts

```diff
@@ -677,6 +707,18 @@ export class Job extends JobBase {
       ...this.checkNoReservedArgs(props.defaultArguments),
     };
 
+    if (props.executionClass === ExecutionClass.FLEX) {
+      if (executable.type !== JobType.ETL) {
+        throw new Error('FLEX ExecutionClass is only available for JobType.ETL jobs');
+      }
+      if ([GlueVersion.V0_9, GlueVersion.V1_0, GlueVersion.V2_0].includes(executable.glueVersion)) {
+        throw new Error('FLEX ExecutionClass is only available for GlueVersion 3.0 or later');
+      }
+      if (props.workerType && (props.workerType !== WorkerType.G_1X && props.workerType !== WorkerType.G_2X)) {
+        throw new Error('FLEX ExecutionClass is only available for WorkerType G_1X or G_2X');
+      }
```

Comment:
In [this page](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-jobs-job.html#aws-glue-api-jobs-job-Job), the following.

`Only jobs with AWS Glue version 3.0 and above and command type glueetl will be allowed to set ExecutionClass to FLEX. The flexible execution class is available for Spark jobs.`

In [this page](https://docs.aws.amazon.com/glue/latest/dg/add-job.html), the following.

`The flexible execution class is appropriate for non-urgent jobs such as pre-production jobs, testing, and one-time data loads. Flexible job runs are supported for jobs using AWS Glue version 3.0 or later and G.1X or G.2X worker types.`